### DISCUSSION TITLE
Install without root permissions

### DISCUSSION THREAD

**Discussion by MicroNateId** [scop/bash-completion#998] [Q&A]
Is there any way to install this to a local user directory without having root permissions?

Work environment is locked down, but I would love to benefit from the full completion capability.

**Comment by akinomyoga** [comment]
You can use the standard `./configure` & `make install` as described in the following part of README.

https://github.com/scop/bash-completion/blob/47aa0a2517070d998f82dc90638632d91990e8be/README.md?plain=1#L32-L43

You can specify the flag `--prefix=/install/path/to/bash-completion` (please replace the path with the one where you want to install bash-completion, such as `$HOME/.local` or `$HOME/.opt/bash-completion/master`) to the `./configure` command in the above instructions.

```bash
./configure --prefix=/install/path/to/bash-completion
```

**Reply by scop** [reply]
Additionally, I'm not sure if we document it anywhere, but it's possible to run in-place from a git clone without actually "installing" anything.

In a nutshell, clone the repo somewhere, `cd` to that dir, then
```shell
autoreconf -i
./configure
make
```
...and just `. /path/to/that/dir/bash_completion` in bashrc.

`make clean && make` is advisable after a git pull later when upgrading to refresh symlinks.

**Comment by MicroNateId** [comment]
Thank, will give that a shot this week - the other bonus is I've got two different linux distributions run with the same home directory mount point... so might need two different folders to configure for what each environment supports, but that doesn't sound too bad.

**Reply by scop** [reply]
Most of the functionality is determined at runtime as opposed to configure/make time, so sharing the same dir between completely different systems should generally work just fine.
